### PR TITLE
improve artifact hashing performance

### PR DESCRIPTION
**Pull request by joprice**

### CODE REVIEW

File: coursier.bzl [hunk 1]
```diff
@@ -552,15 +556,25 @@ def _coursier_fetch_impl(repository_ctx):
...
+
+    exec_result = repository_ctx.execute(hasher_command)
+    if exec_result.return_code != 0:
+        fail("Error while obtaining the sha256 checksums: " + exec_result.stderr)
+
+    shas = {}
+    for line in exec_result.stdout.splitlines():
+      parts = line.split(" ")
+      path = repository_ctx.path(parts[1])
+      shas[path] = parts[0]
```
Comment (jin) [inline coursier.bzl]: use `artifact["coord"]` as the key? that should solve the CI failure.

**Comment by jin** [comment]
and thank you so much for this!

File: coursier.bzl (hunk 1 above)
Comment (joprice) [reply coursier.bzl]: I didn't realize path wasn't hashable on some versions. How do you suggest going about that for coord? Pass a series of concatenations of file and coord to the java process like `coord:file coord:file`, then return `coord:sha coord:sha`?

**Comment by joprice** [comment]
I'll remove the mods file. It was leftover from a series of optimization attempts I made using graal, newer jdk, java modules, etc to see if I could shave off more of the startup time. They all ended up being about the same or in the case of graal, worse.

File: coursier.bzl (hunk 1 above)
Comment (jin) [reply coursier.bzl]: Hmm, you could also cast path to `str()`. Should be a fairly fast opertation.

File: coursier.bzl [hunk 2]
```diff
@@ -508,16 +519,17 @@ def _coursier_fetch_impl(repository_ctx):
         primary_url_parts = []
         filepath_parts = artifact["file"].split("/")
         protocol = None
+
         # Only support http/https transports
         for part in filepath_parts:
             if part == "http" or part == "https":
-                    protocol = part
+                protocol = part
```
Comment (jin) [inline coursier.bzl]: I believe you need to rebase this part on master to add `break` after this line.

File: coursier.bzl (hunk 1 above)
Comment (joprice) [reply coursier.bzl]: Ok I'll try that first

**Pull request merged**